runtime/jvmti/jvmtiExtensionMechanism.c

```diff
@@ -1911,22 +1956,26 @@ iterateSharedCachesCallback(J9JavaVM *vm, J9SharedCacheInfo *event_data, void *u
  * user_data - opaque user-supplied pointer which will be passed to their callback.
  */ 
 static jvmtiError JNICALL
-jvmtiIterateSharedCaches(jvmtiEnv *env,
-	jint version,
-	const char *cacheDir,
-	jint flags,
-	jboolean useCommandLineValues,
-	jvmtiIterateSharedCachesCallback callback,
-	void *user_data, ...)
+jvmtiIterateSharedCaches(jvmtiEnv *env, ...)
 {
 	J9JavaVM *vm = JAVAVM_FROM_ENV(env);
 	jvmtiError rc = JVMTI_ERROR_NOT_AVAILABLE;
 #if defined(J9VM_OPT_SHARED_CLASSES)
 	struct IterateSharedCacheUserdata userdata;
 #endif
-	
+
+	va_list args;
+	va_start(args, env);
+	jint version = va_arg(args, jint);
+	const char *cacheDir = va_arg(args, char *);
+	jint flags = va_arg(args, jint);
+	jboolean useCommandLineValues = (jboolean)va_arg(args, jint);
```

Comment:
I first wrote this line as `jboolean useCommandLineValues = va_arg(args, jboolean);`, but I got a compiler error on AArch64 macOS as follows:

```
error: second argument to 'va_arg' is of promotable type 'jboolean' (aka 'unsigned char'); this va_arg has undefined behavior because arguments will be promoted to 'int' [-Werror,-Wvarargs]
```

Reply:
Same issue I was having with my general solution.  Are we certain that all platforms will pass variadic arguments as at least 32 bits?

Reply:
I need to think more about this.  No conclusion at the moment.

Reply:
Passing a `jboolean` value as one of variable length arguments will always promote it to `int`, according to "4.8.1.2 The va_arg macro" in ANSI C Rationale:

> When using these macros it is important to remember that the type of an argument in a variable argument list will never be an integer type smaller than int, nor will it ever be float.

I changed the code as `(jboolean)va_arg(args, int)` instead of `(jboolean)va_arg(args, jint)` for in case where `jint` is different from native `int`.